### PR TITLE
Enable the modernize-loop-convert check on clang-tidy

### PR DESCRIPTION
**Pull request by er-vin**
Signed-off-by: Kevin Ottens <kevin.ottens@nextcloud.com>

### CODE REVIEW

File: src/gui/folderstatusmodel.cpp [hunk 1]
```diff
@@ -413,8 +413,8 @@ FolderStatusModel::SubFolderInfo *FolderStatusModel::infoForIndex(const QModelIn
 FolderStatusModel::SubFolderInfo *FolderStatusModel::infoForFileId(const QByteArray& fileId, SubFolderInfo* info) const
 {
   const QVector<SubFolderInfo>& infoVec = info ? info->_subs : _folders;
-  for(int i = 0, end = infoVec.size(); i < end; i++) {
-    auto *info = const_cast<SubFolderInfo *>(&infoVec[i]);
+  for(const auto &i : infoVec) {
+    auto *info = const_cast<SubFolderInfo *>(&i);
```
Comment (sbeyer) [inline src/gui/folderstatusmodel.cpp:417]: This function totally creeps me out for several reasons... Moreover, it is never used. Any objections to simply remove this method (for clean history in a separate commit before the modernizing loop commit)?

File: src/libsync/cookiejar.cpp [hunk 2]
```diff
@@ -34,8 +34,8 @@ QDataStream &operator<<(QDataStream &stream, const QList<QNetworkCookie> &list)
 {
     stream << JAR_VERSION;
     stream << quint32(list.size());
-    for (int i = 0; i < list.size(); ++i)
-        stream << list.at(i).toRawForm();
+    for (const auto &cookie : list)
+        stream << cookie.toRawForm();
```
Comment (sbeyer) [inline src/libsync/cookiejar.cpp:38]: Yay, a sensible variable name \o/

File: src/libsync/owncloudpropagator.cpp [hunk 3]
```diff
@@ -796,16 +796,16 @@ bool PropagatorCompositeJob::scheduleSelfOrChild()
     }
 
     // Ask all the running composite jobs if they have something new to schedule.
-    for (int i = 0; i < _runningJobs.size(); ++i) {
-        ASSERT(_runningJobs.at(i)->_state == Running);
+    for (auto runningJob : qAsConst(_runningJobs)) {
```
Comment (sbeyer) [inline src/libsync/owncloudpropagator.cpp:799]: Why not by const-reference here?

File: src/gui/folderstatusmodel.cpp (hunk 1 above)
Comment (er-vin) [reply src/gui/folderstatusmodel.cpp:417]: Yeah, don't like that method either... And indeed unused, good catch. I'll kill it.

File: src/libsync/cookiejar.cpp (hunk 2 above)
Comment (er-vin) [reply src/libsync/cookiejar.cpp:38]: Yes... I know, right? :-)

File: src/libsync/owncloudpropagator.cpp (hunk 3 above)
Comment (er-vin) [reply src/libsync/owncloudpropagator.cpp:799]: Because this is a scalar type, there's nothing to gain from the const-ref.

Comment (sbeyer) [reply src/libsync/owncloudpropagator.cpp:799]: If that was the case, I wouldn't have asked :-) But `PropagatorJob` doesn't look very scalar to me.

Comment (er-vin) [reply src/libsync/owncloudpropagator.cpp:799]: `PropagatorJob *` on the other hand. ;-)

Comment (sbeyer) [reply src/libsync/owncloudpropagator.cpp:799]: ....looks very scalar. Good :) Acknowledged.

**Pull request merged**